app/src/main/java/com/ataulm/artcollector/gallery/ui/GalleryActivity.kt

```diff
@@ -48,14 +50,8 @@ class GalleryActivity : AppCompatActivity() {
     }
 
     private fun navigateToPainting(command: NavigateToPainting) {
-        val (painting, adapterPosition) = command.painting to command.adapterPosition
-        val paintingIntent = paintingIntent(painting.artistId, painting.id, painting.imageUrl)
-        val options = ActivityOptionsCompat.makeSceneTransitionAnimation(this, recyclerView.sharedElements(adapterPosition))
+        val paintingIntent = paintingIntent(command.artistId, command.paintingId, command.imageUrl)
+        val options = ActivityOptionsCompat.makeSceneTransitionAnimation(this, Pair(command.view, getString(R.string.shared_element_painting)))
```

Comment:
these changes aren't paging relevant, just de-tangles some of the UiModel from the events.